cluster-autoscaler/config/autoscaling_options.go

```diff
@@ -186,6 +186,8 @@ type AutoscalingOptions struct {
 	MaxPodEvictionTime time.Duration
 	// IgnoredTaints is a list of taints to ignore when considering a node template for scheduling.
 	IgnoredTaints []string
+	// UnwantedTaints is a list of taints to remove when creating a node template for scheduling.
+	UnwantedTaints []string
```

Comment:
I really don't like this name, it only make sense in context of how the variable is used in the code, not what the variable wants to achieve (ie. name is driven by implementation not use-case).
I'd consider calling those StatusTaints, StateTaints or ConditionTaints or similar. I think that's a good name for non-permanent taint that can be added during lifetime of the node, because any such taint would presumably represent some sort of node state. And the name doesn't say how we're treating the taint in CA logic, instead focusing on what the taint is meant to represent.